FrontDesk/tests/UnitTests/Core/AggregatesEntities/AppointmentTests/Appointment_Confirm.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Linq;
+using AutoFixture;
+using FrontDesk.Core.Aggregates;
+using FrontDesk.Core.Events;
+using Xunit;
+
+namespace UnitTests.Core.AggregatesEntities.AppointmentTests
+{
+  public class Appointment_Confirm
+  {
+    private Fixture _fixture = new Fixture();
+
+    [Fact]
+    public void ReconfirmResultDateNotChanged()
+    {
+      var appointment = _fixture.Build<Appointment>()
+        .Without(z => z.Events)
+        .Create();
+
+      var confirmedDate = DateTime.Now;
+
+      appointment.Confirm(confirmedDate);
+
+      Assert.NotEqual(confirmedDate, appointment.DateTimeConfirmed);
+    }
+
+    [Fact]
+    public void ConfirmFirstTimeResultConfirmDateUpdated()
+    {
+      var appointment = _fixture.Build<Appointment>()
+        .Without(z => z.Events)
+        .Create();
+      appointment.DateTimeConfirmed = null;
+
+      var confirmedDate = DateTime.Now;
+
+      appointment.Confirm(confirmedDate);
+
+      Assert.Equal(confirmedDate, appointment.DateTimeConfirmed);
+    }
+
+    [Fact]
+    public void ConfirmFirstTimeResultEventCreated()
+    {
+      var appointment = _fixture.Build<Appointment>()
+        .Without(z => z.Events)
+        .Create();
+      appointment.DateTimeConfirmed = null;
+
+      var confirmedDate = DateTime.Now;
+
+      appointment.Confirm(confirmedDate);
+
+      Assert.NotNull(appointment.Events.FirstOrDefault(x => x.GetType() == typeof(AppointmentConfirmedEvent)));
```

Comment:
You can use the `Contains` assertion for this and it's a bit cleaner I think. See my example in Appointment_UpdateRoom.